### PR TITLE
Run CI on Windows

### LINKED ISSUE #20: Run the native build github action on Windows
We previously had an issue with running github actions on the windows image but that shouldn't be an issue anymore

### PR DESCRIPTION
**Pull request by stefunctional**
### Issues:

Resolves #20 

### Description of changes:

Adapt github action to run on Windows.

### Call-outs:

N/A

### Testing:

CI passed on all platforms.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT license.

### CODE REVIEW

File: mls-rs-crypto-awslc/Cargo.toml [hunk 1]
```diff
@@ -20,10 +20,10 @@ zeroize = { version = "1", features = ["zeroize_derive"] }
 maybe-async = "0.2.7"
 
 [target.'cfg(all(target_arch = "aarch64", macos))'.dependencies]
-aws-lc-sys = { version = "0.11.0", features = ["bindgen"] }
+aws-lc-sys = { version = "0.12.0", features = ["bindgen"] }
 
 [target.'cfg(windows)'.dependencies]
-aws-lc-sys = { version = "0.11.0", features = ["bindgen"] }
+aws-lc-sys = { version = "0.12.0", features = ["bindgen"] }
```
Comment (tomleavy) [inline mls-rs-crypto-awslc/Cargo.toml:26]: Can you change this to add bindgen when not one of the default supported arch? That will help with android + iOS

all(target_os = "macos", target_arch = "x86_64"),
                all(target_os = "linux", target_arch = "x86"),
                all(target_os = "linux", target_arch = "x86_64"),
                all(target_os = "linux", target_arch = "aarch64")

Comment (stefunctional) [reply mls-rs-crypto-awslc/Cargo.toml:26]: Can we actually get rid of all lines `aws-lc-sys = ... features = ["bindgen"] ...` in our `Cargo.toml`? As you mentioned during a meeting, `aws-lc-sys` already [forces](https://github.com/aws/aws-lc-rs/blob/cb030a33fb22cba4441f2b2d17e3c1c702c39895/aws-lc-sys/Cargo.toml#L60-L61) the use of `bindgen` for platforms that need it. If one of our users wants to use `bindgen` to build `aws-lc-sys` on their system (even though their architecture is supported without `bindgen`), they can do so by adding a dependency on `aws-lc-sys` to their `Cargo.toml` and enable the `bindgen` feature there. What do you think?

Comment (tomleavy) [reply mls-rs-crypto-awslc/Cargo.toml:26]: Yeah, if it works fine without these lines we can remove them.

Comment (stefunctional) [reply mls-rs-crypto-awslc/Cargo.toml]: I removed them and CI passed for Windows.

**Review by tomleavy — APPROVED** [review]

**Review by mulmarta — APPROVED** [review]

**Pull request merged**